### ISSUE TITLE
UAWebViewClient not handling https://device-api.urbanairship.com/favicon.ico

### ISSUE THREAD

**Issue by randomaspirer** [urbanairship/android-library#117]
# Preliminary Info

### What Urban Airship dependencies are you using?
ℹ Please include Urban Airship dependencies and their versions here.
Android UA SDK with FCM
implementation 'com.urbanairship.android:urbanairship-fcm:9.7.1'

### What are the versions of any relevant development tools you are using?
ℹ If applicable, please include any relevant information about the version of your
IDE, plugin(s), or package manager(s) here.

AndroidStudio 3.3.1

# Report

### What unexpected behavior are you seeing?
ℹ Please provide a brief description of the unexpected behavior here.
UAWebView client giving error for `WebResourceRequest.getUrl = https://device-api.urbanairship.com/favicon.ico` and sending the error callback to `UAWebview.UAWebViewClient's onReceivedError(WebView, WebResourceRequest, WebResourceError)` method with following error
WebResourceError.getErrorCode() = -1 = WebViewClient.ERROR_UNKNOWN
WebResourceError.getDescription() = There was a network error.

### What is the expected behavior?
ℹ Please provide a brief description of the expected behavior here.
UAWebViewclient shouldInterceptRequest() method should handle the https://device-api.urbanairship.com/favicon.ico 
I know this was mentioned in https://github.com/urbanairship/android-library/issues/97 but that didn't solve it and propagates the error to client

### What are the steps to reproduce the unexpected behavior?
ℹ Please provide reproduction steps here.
Try to load a Rich Inbox message in message center with a UAWebViewClient set on a UAWebView and overriding `onReceivedError(WebView, WebResourceRequest, WebResourceError)` you will get error callback for url https://device-api.urbanairship.com/favicon.ico with 
```
WebResourceError.getErrorCode() = -1 = WebViewClient.ERROR_UNKNOWN
WebResourceError.getDescription() = There was a network error.
``` 
which should be consumed by UAWebViewClient()

I am for now checking in onReceivedError of my customview for what you guys do in shouldInterceptRequest() by looking for path that ends with "/favicon.ico", it would be better if you could handle this in UAWebViewClient() shouldIntercepRequest() in a differentway that would solve this. The previous versions of the sdk without the change of shouldInterceptRequest() code in UAWebViewClient works fine.
### Do you have logging for the issue?
ℹ Please provide links to gists of any pertinent logs here.

**Comment by BrianBatchelder** [comment]
Thanks for the report. We'll look into this ASAP.

**Comment by rlepinski** [comment]
Our change prevents the network request but at the cost of the error. For now you can ignore those errors if the request URL ends in `/favicon.ico`. We are looking for a better solution that prevents the web request and error. Looks like that might mean shipping an empty image that we can return as the favicon instead of null.

**Comment by randomaspirer** [comment]
Okay our team will keep track on this issue for any update you guys make, thanks for responding swiftly.

**Comment by rlepinski** [comment]
Should be fixed in 10.0.0
